### PR TITLE
[FIX] website: handle non website_form compatible models

### PR DESCRIPTION
**Pull request by ushyme**
### Steps to reproduce

* install `website_sale`
* in the settings, enable 'Extra Step During Checkout'
* go to the Extra Info step in the checkout process
* switch to edit mode and click on any input in the form

You should be met with a traceback: "Cannot read property of undefined (reading 'website_form_label')"

### Cause

This issue was introduced with odoo/odoo@3626e36a9c4995286be48206b0d927f1de51e295
Basically, if you try to edit a form whose model is not one of the `compatible_form_models`, you get a traceback because the system attempts to access `website_form_label` on an empty form.

opw-3891255

### CODE REVIEW

**Comment by rdeodoo** [comment]
@robodoo delegate=@bso-odoo

File: addons/website/static/src/snippets/s_website_form/options.js [hunk 1]
```diff
@@ -366,7 +366,7 @@ options.registry.WebsiteFormEditor = FormEditor.extend({
         
         const targetModelName = this.$target[0].dataset.model_name || 'mail.mail';
         this.activeForm = this.models.find(m => m.model === targetModelName);
-        currentActionName = this.activeForm.website_form_label;
+        currentActionName = this.activeForm && this.activeForm.website_form_label;
```
Comment (bso-odoo) [inline addons/website/static/src/snippets/s_website_form/options.js:369]: Indeed, we should have moved that line inside the `if (!this.modelCantChange)`.
But with your fix, it should also avoid failing for forms where `hide-change-model` has not been specified - until an action is selected.

**Review by bso-odoo — APPROVED** [review]

**Comment by bso-odoo** [comment]
@robodoo r+
